src/volatile_memory.rs

```diff
@@ -588,6 +588,10 @@ impl<B: BitmapSlice> Bytes<usize> for VolatileSlice<'_, B> {
     /// assert_eq!(res.unwrap(), 4);
     /// ```
     fn write(&self, buf: &[u8], addr: usize) -> Result<usize> {
+        if self.size == 0 {
```

Comment:
Hmm, I think we should check instead the size of buffer, not of self here.

Reply:
Yes so if the size of the initial slice is 0, I think we should actually fail for whatever operation we are trying to do to that slice.

Reply:
Updated it.